packages/@ember/-internals/routing/lib/system/route.ts

```diff
@@ -383,17 +383,17 @@ class Route extends EmberObject implements IRoute {
 
     ```app/routes/contact-form.js
     import Route from '@ember/routing/route';
-
-    export default Route.extend({
-      actions: {
-        willTransition(transition) {
-          if (this.controller.get('userHasEnteredData')) {
-            this.controller.displayNavigationConfirm();
-            transition.abort();
-          }
+    import { action } from '@ember/object';
+
+    export default class ContactFormRoute extends Route {
+      @action
+      willTransition(transition) {
```

Comment:
I'm slightly concerned about this one. Specifically, `Route` has both an action _and_ a method named `willTransition`. The `@action` decorator will result in both with be populated (with the same function). 

@pzuraq - Will this result in a loop?

Reply:
I don't think it would result in a loop, it'll just be that the function gets called from two places instead of one, unless the `willTransition` function also calls the action

Reply:
I'm actually not seeing that the method exists, it appears to just be the action/event.